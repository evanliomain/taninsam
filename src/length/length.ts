/**
 * @module array=>number
 */
/**
 * Get the length of the table
 * @return the function to apply to get the array length
 * @example
 * ```
 * length<number>()([0, 1, 2, 3, 4]) // 5
 * ```
 * @example Using the chain
 * ```
 * chain([0, 1, 2, 3, 4])
 *   .chain(length<number>())
 *   .value() // 5
 * ```
 */
export function length<T>(): (elements: ReadonlyArray<T>) => number {
  return (elements: ReadonlyArray<T>) => elements.length;
}
