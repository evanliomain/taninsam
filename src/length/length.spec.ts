import { describe, expect, test } from 'vitest';
import { length } from './length';

describe('length function', () => {
  const input = [1, 2, 3, 4, 5];

  test('is a pure function', () => {
    length()(input);
    expect(input).toMatchSnapshot();
  });

  test('input |> length === 5', () => {
    expect(length()(input)).toBe(5);
  });

  test('[] |> length === 0', () => {
    expect(length()([])).toBe(0);
  });
});
