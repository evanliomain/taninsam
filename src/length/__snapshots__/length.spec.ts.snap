// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`length function is a pure function 1`] = `
Array [
  1,
  2,
  3,
  4,
  5,
]
`;
